src/lib/utils/functions/EntityTypeUtils.ts

```diff
@@ -289,3 +293,70 @@ export function getVersionDisplay(entity: Entity): string {
     return entity.versionNumber!.toString()
   }
 }
+
+// implemented by https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/Entity.html
+const allEntityKeys: string[] = [
+  'name',
+  'description',
+  'id',
+  'etag',
+  'createdOn',
+  'modifiedOn',
+  'createdBy',
+  'modifiedBy',
+  'parentId',
+  'concreteType',
+]
+// https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/VersionableEntity.html
+const versionableKeys: string[] = [
+  ...allEntityKeys,
+  'versionNumber',
+  'versionLabel',
+  'versionComment',
+  'isLatestVersion',
+]
+
+// https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/table/Table.html
+const tableKeys: string[] = [...versionableKeys, 'columnIds', 'isSearchEnabled']
+
+// https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/table/View.html
+const viewKeys: string[] = [...tableKeys]
+
+// https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/table/EntityRefCollectionView.html
+const entityRefCollectionViewKeys: string[] = [...viewKeys, 'items']
+
+/**
+ * A string array of all possible keys used by Synapse in Entity objects (objects that inherit this interface: https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/Entity.html).
+ * This object is used to determine which fields are standard and which are annotations,
+ * so it's important that this object contains all keys in the objects that implement the linked interface above.
+ */
+export const entityJsonKeys: Record<ENTITY_CONCRETE_TYPE, string[]> = {
```

Comment:
Change from a list of keys to a mapping from concreteType -> keys

Reply:
Any way to leverage your entity interface definitions for these types?  It's unfortunate that the fields need to be defined twice in the codebase.